src/pkg/shub/client/pull_test.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package client
+
+import (
+	"os"
+	"testing"
+
+	"github.com/singularityware/singularity/src/pkg/test"
+)
+
+const (
+	shubURI     = "shub://ikaneshiro/singularityhub:latest"
```

Comment:
Can we test retrieval using the `@digest` for as well, since we are supporting that in the URI scheme?

Reply:
Will issue it as a TODO, it will add more download steps to the test process